### ISSUE TITLE
Snap package on snap store is outdated

### ISSUE THREAD

**Issue by KhazAkar** [Ericsson/codechecker#4276]
**Describe the bug**
Last published version of code checker is 6.21, which was released on snap store 2022-11-11

**CodeChecker version**
N/A

**To Reproduce**
1. `snap info codechecker`
OR
1. https://snapcraft.io/codechecker

**Expected behaviour**
Updated  automatically per tagged release on github

**Desktop (please complete the following information)**
 - OS: Ubuntu 22.04/24.04
 - Browser: Firefox

**Additional context**
Those links might help with automatic releases on snap store :)
https://github.com/snapcore/action-build
https://github.com/snapcore/action-publish 
https://github.com/marketplace/actions/snapcraft-action

**Comment by whisperity** [comment]
Also, is there anything significant that the *snap* distribution provides that can't be done by using the PyPI release (which, I know, is also outdated for now, see #4261)? I would prefer if there weren't 80 separate release pipelines with 80 separate chances of getting broken or incompatible...

**Comment by KhazAkar** [comment]
> Also, is there anything significant that the _snap_ distribution provides that can't be done by using the PyPI release (which, I know, is also outdated for now, see #4261)? I would prefer if there weren't 80 separate release pipelines with 80 separate chances of getting broken or incompatible...

With rolling out things like managed python, see `externally-managed-environment`, it's not as convenient as just `snap install codechecker --classic` to get the ball rolling quickly.

For automatic releases, I might take a look at code to see what's going on, maybe it's trivial, just overlooked thing :)

**Comment by whisperity** [comment]
Yes, we are planning to harden the releases in a way that even if it is not **automatic** per sē, it should generate something automatically that can undergo manual testing and then we can flick a switch to re-run the job to deploy automatically.

However, I can also see that the Snap package is attributed to (and potentially owned by) @csordasmarton. So this will require some more in-depth investigation...

**Comment by KhazAkar** [comment]
For such things, it's best to create topic on forum.snapcraft.io .
@csordasmarton can add you as co-developer and potentially transfer package to you via forum post.

**Comment by csordasmarton** [comment]
@bruntib and @vodorok were added as collaborators to the codechecker package so they should be able to manage it as well.

**Comment by steakhal** [comment]
Snap is still outdated.
